src/ralph/data_center/api/serializers.py

```diff
@@ -123,6 +125,34 @@ class DataCenterAssetSerializer(ComponentSerializerMixin, AssetSerializer):
     scmstatuscheck = SCMInfoSerializer()
     securityscan = SecurityScanSerializer()
 
+    related_hosts = serializers.SerializerMethodField()
+
+    def _get_physical_hosts(self, obj):
+        physical_server = ContentType.objects.get_for_model(DataCenterAsset)
+        return DataCenterAssetSimpleSerializer(
+            obj.children.filter(content_type=physical_server), many=True
```

Comment:
when you use filter, you are not using any prefetched data (by `prefetch_related` call) and you make here call(s) to the database foreach DataCenterAsset you're processing - I think you should filter out these objects in your code (not in database) or build prefetch new, filtered objects (for all `DataCenterAsset`s at once)

Reply:
Done.